the-collab-lab/tcl-17-smart-shopping-list
# VT DD list add routes

## Description


This PR adds three components: one for list view, add item view, and navigation.

### Add dependencies for react router dom
`npm install --save react-router-dom`

### Navigation tabs
Clicking navigation links updates to the list or add item views. The text in tab displays in bold and the background color changes.

### Route NavLink Switch
Components of react router dom. Route renders parts of the UI, NavLink allows navigation without a full refresh and also styling attributes, Switch renders a route exclusively.
More documentation on `react-router-dom` [here](https://reactrouter.com/core/guides/quick-start)

## Related Issue


Closes [#3](https://github.com/the-collab-lab/tcl-17-smart-shopping-list/issues/3)

## Acceptance Criteria


- [ ] The following has been added as a project dependency: `react-router-dom`

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|  ✓ | :link: Update dependencies |

## Updates

<img width="513" alt="list_view" src="https://user-images.githubusercontent.com/14856004/103716630-28e2d400-4f89-11eb-8088-768adc9cfb8b.png">
<img width="513" alt="add_view" src="https://user-images.githubusercontent.com/14856004/103716642-31d3a580-4f89-11eb-8c12-709e8d1abb95.png">


## Testing Steps / QA Criteria



* From your terminal, pull down this branch with `git pull origin VT-DD-list-add-routes` and check that branch out with `git checkout VT-DD-list-add-routes`
* Then `npm install --save react-router-dom`  to install the newly added dependencies locally and `npm start` to launch the app.
* Or as Lauren suggested but dark mode!
<img width="840" alt="Screen Shot 2021-01-05 at 7 37 17 PM" src="https://user-images.githubusercontent.com/14856004/103719168-0f905680-4f8e-11eb-8e74-c2663b965833.png">

* Click the list or add an item button change views
